src/node/index.d.ts

```diff
@@ -0,0 +1,526 @@
+
+    /* Unrecoverable data loss or corruption. */
+    DATA_LOSS = 15,
+
+    /* Force users to include a default branch: */
+    _DO_NOT_USE = -1
+  }
+
+  /**
+   * Propagate flag name to number mapping
+   */
+  export const propagate: {}
+
+  /**
+   * Call error name to code number mapping
+   */
+  export const callError: {}
+
+  /**
+   * Write flag name to code number mapping
+   */
+  export const writeFlags: {}
+
+  /**
+   * Log verbosity setting name to code number mapping
+   */
+  export const logVerbosity: {}
+
+  /**
+   * Creates a constructor for a client with the given methods.
+   * @param methods An object mapping method names to method attributes
+   * @param serviceName The fully qualified name of the service
+   * @return New client constructor
+   */
+  export function makeGenericClientConstructor(methods: IMethodsMap, serviceName: string): Client;
+
+  /**
+   * Create a client with the given methods
+   */
+  export class Client {
+    /**
+     * Create an instance of Client
+     * @param address The address of the server to connect to
+     * @param credentials Credentials to use to connect to the server
+     * @param options Options to pass to the underlying channel
+     */
+    constructor(address: string, credentials: any, options: IClientOptions)
+  }
+
+  export interface IClientOptions {
+
+  }
+
+  /**
+   * The methods object map.
+   */
+  export interface IMethodsMap {
+    [index: string]: {
+      /**
+       * The path on the server for accessing the method. For example, for protocol buffers, we use "/service_name/method_name"
+       */
+      path: string;
+
+      /**
+       * Indicating whether the client sends a stream
+       */
+      requestStream: boolean;
+
+      /**
+       * Indicating whether the server sends a stream
+       */
+      responseStream: boolean;
+
+      /**
+       * Function to serialize request objects
+       */
+      requestSerialize: () => any;
+
+      /**
+       * Function to deserialize response objects
+       */
+      responseDeserialize: () => any;
+    }
+  }
+
+  /**
+   * Return the underlying channel object for the specified client
+   * @param client
+   * @return The channel
+   */
+  export function getClientChannel(client: Client): any;
+
+  /**
+   * Wait for the client to be ready. The callback will be called when the
+   * client has successfully connected to the server, and it will be called
+   * with an error if the attempt to connect to the server has unrecoverablly
+   * failed or if the deadline expires. This function will make the channel
+   * start connecting if it has not already done so.
+   * @param client The client to wait on
+   * @param deadline When to stop waiting for a connection. Pass Infinity to wait forever.
+   * @param callback The callback to call when done attempting to connect.
+   */
+  export function waitForClientReady(client: Client, deadline: Date | number, callback: (error?: Error) => void): void;
```

Comment:
This should likely be `error: Error | null` instead of `error?: Error`, which is equivalent to `error: Error | undefined`. I haven't checked with grpc specifically, but all stdlib Node APIs use `null` in this situation.